### DISCUSSION TITLE
using NextApiRequest with query?

### DISCUSSION THREAD

**Discussion by stoplion** [vercel/next.js#34044] [Help]
```
export default function handler(
  req: NextApiRequest,
  res: NextApiResponse<Data>
) {

  const searchTerm: string = req.query.searchTerm
  ......
```

Typescript error: Type 'string | string[]' is not assignable to type 'string'. Type 'string[]' is not assignable to type 'string'.

How to make NextApiRequest play nice with Typescript when using query?

**Comment by wwwdepot** [comment]
This is probably what you looking for
```ts
const searchTerm = req.query.searchTerm as string;
```

BUT it's possible to pass multiple query with same name in the URL... e.g.: 

`/api/search?q=1&q=2&q=3` 👉 `q = ["1", "2", "3"]`

`/api/search?q=1` 👉 `q = "1"`

What will happen when you use an array instead of a string?

**Reply by omriluz** [reply]
Thanks a lot!!! it worked for me
